google/cloud/testing_util/mock_sources.h

```diff
@@ -0,0 +1,79 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_TESTING_UTIL_MOCK_SOURCES_H
+#define GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_TESTING_UTIL_MOCK_SOURCES_H
+
+#include "google/cloud/future.h"
+#include "absl/types/variant.h"
+#include <chrono>
+#include <deque>
+#include <thread>
+
+namespace google {
+namespace cloud {
+inline namespace GOOGLE_CLOUD_CPP_NS {
+namespace testing_util {
+
+/**
+ * A type meeting the requirements for source<T, E> used in testing.
+ *
+ * @tparam T the value type for the source
+ * @tparam E the error type for the source
+ */
+template <class T, typename E>
+class MockSource {
```

Comment:
Naming nit: The word "mock" here may make readers think that this object is doing some verification (it's not, is it?). Especially with our use of Googletest, readers may expect it to respond to ON_CALL() and friends.

I think "Fake" may be a more commonly used term for this type of class. WDYT?

Reply:
Fixed.